x86.Dockerfile

```diff
@@ -1,19 +1,81 @@
 FROM tensorflow/tensorflow:latest-py3
 
-RUN apt-get update && apt-get install -y pkg-config libsm6 libxext6 libxrender-dev
+# The `python3-opencv` package isn't built with gstreamer on Ubuntu. So we need to manually build opencv.
+ARG OPENCV_VERSION=4.3.0
+# http://amritamaz.net/blog/opencv-config
+RUN apt-get update && apt-get install -y --no-install-recommends \
+        build-essential \
+        cmake \
+        curl \
+        git \
+        gstreamer1.0-plugins-bad \
+        gstreamer1.0-plugins-good \
+        gstreamer1.0-plugins-ugly \
+        gstreamer1.0-vaapi \
+        libavcodec-dev \
+        libavformat-dev \
+        libgstreamer-plugins-base1.0-dev \
+        libgstreamer1.0-dev \
+        libsm6 \
+        libswscale-dev \
+        libxext6 \
+        libxrender-dev \
+        mesa-va-drivers \
+        python3-dev \
+        python3-numpy \
+    && rm -rf /var/lib/apt/lists/* \
+    && cd /tmp/ \
+    && curl -L https://github.com/opencv/opencv/archive/${OPENCV_VERSION}.tar.gz -o opencv.tar.gz \
+    && tar zxvf opencv.tar.gz && rm opencv.tar.gz \
+    && cd /tmp/opencv-${OPENCV_VERSION} \
+    && mkdir build \
+    && cd build \
+    && cmake \
+        -DBUILD_opencv_python3=yes \
+        -DPYTHON_EXECUTABLE=$(which python3) \
+        -DCMAKE_BUILD_TYPE=RELEASE \
+        -DBUILD_TESTS=OFF \
+        -DBUILD_PERF_TESTS=OFF \
+        -DBUILD_EXAMPLES=OFF \
+        -DINSTALL_TESTS=OFF \
+        -DBUILD_opencv_apps=OFF \
+        -DBUILD_DOCS=OFF \
+        ../ \
+    && make -j$(nproc) \
+    && make install \
+    && cd /tmp \
+    && rm -rf opencv-${OPENCV_VERSION} \
+    && apt-get purge -y \
+        cmake \
+        git \
+        libgstreamer-plugins-base1.0-dev \
+        libgstreamer1.0-dev \
+        libxrender-dev \
+        python3-dev \
+    && apt-get autoremove -y
 
-# get all pip packages
-RUN pip install --upgrade pip setuptools==41.0.0 && pip install \
-    aiofiles \
-    fastapi \
-    image \
-    scipy \
-    uvicorn \
-    wget \
-    opencv-python \
-    pyzmq
+# https://askubuntu.com/questions/909277/avoiding-user-interaction-with-tzdata-when-installing-certbot-in-a-docker-contai
+ARG DEBIAN_FRONTEND=noninteractive
 
-ENTRYPOINT ["python", "neuralet-distancing.py"]
+RUN apt-get update && apt-get install -y --no-install-recommends \
+        pkg-config \
+        python3-dev \
+        python3-numpy \
+        python3-pillow \
+        python3-pip \
+        python3-scipy \
+        python3-wget \
+    && rm -rf /var/lib/apt/lists/* \
+    && python3 -m pip install --upgrade pip setuptools==41.0.0 && pip install \
+        aiofiles \
+        fastapi \
+        uvicorn \
+    && apt-get purge -y \
+        python3-dev \
+    && apt-get autoremove -y
+
+
+ENTRYPOINT ["python3", "neuralet-distancing.py"]
 CMD ["--config", "config-x86.ini"]
```

Comment:
I built this docker successfully but it didn't show the video when I ran it. 
It even wrote this report:
INFO:libs.core:processed frame 1001 for /repo/test/data/TownCentreXVID.avi

Reply:
had same issue on x86-openvino docker

Reply:
I guess you have been pulling an incompatible frontend image (https://github.com/neuralet/smart-social-distancing/blob/d043fd2e9077ab039efc80cd4890ef2006bf484e/x86.Dockerfile#L83) from dockerhub. For branches that have different frontend, like this branch, we need to build the frontend image, rather than pulling it. Once we merge this branch to master and publish the new docker image for frontend, you won't require building manually.